lib/OnyxCache.js

```diff
@@ -179,12 +179,23 @@ class OnyxCache {
      * Remove keys that don't fall into the range of recently used keys
      */
     removeLeastRecentlyUsedKeys() {
-        while (this.recentKeys.size > this.maxRecentKeysSize) {
-            const iterator = this.recentKeys.values();
+        let toRemove = this.recentKeys.size - this.maxRecentKeysSize;
+        if (toRemove <= 0) {
+            return;
+        }
+        const iterator = this.recentKeys.values();
+        const temp = [];
+        while (toRemove > 0) {
             const value = iterator.next().value;
             if (value !== undefined) {
-                this.drop(value);
+                temp.push(value);
             }
+            toRemove--;
+        }
+
+        for (let i = 0; i < temp.length; ++i) {
```

Comment:
Why aggregate `temp` in a separate array? Could we just do these two lines inline instead of in a separate loop?

```diff
- temp.push(value);
+ delete this.storageMap[value];
+ this.recentKeys.delete(value);
```

Reply:
because it modifies the bst and most likely breaks iterator